### PR TITLE
Add KeyConfig::derive API.

### LINKED ISSUE #14: Derive KeyConfigs
It would be incredibly useful to be able to (1) generate and then de/serialize private key material for persistence across restarts, or (2) derive key material from a fixed seed. Perhaps one of these could be added to the `KeyConfig` API in the `ohttp` create?

### PR DESCRIPTION
**Pull request by chris-wood**
I didn't add an "unimplemented" error for the NSS variant as I'm not sure what's considered best practice for that. (Do we just panic? Carve out an error? Something else?) But the rust-hpke path is fairly straightforward.

Closes #14.

### CODE REVIEW

**Comment by chris-wood** [comment]
(I'll sort out the NSS build issue later. Demoting to draft PR in the meantime.)

File: ohttp/src/lib.rs [hunk 1]
```diff
@@ -116,6 +116,23 @@ impl KeyConfig {
         })
     }
 
+    /// Derive a configuration for the server side from input keying material
+    /// # Panics
+    /// If the configurations don't include a supported configuration.
+    #[cfg(feature = "rust-hpke")]
+    pub fn derive(key_id: u8, kem: Kem, mut symmetric: Vec<SymmetricSuite>, ikm: &[u8]) -> Res<Self> {
```
Comment (martinthomson) [inline ohttp/src/lib.rs:123]: I think that you want this to be unconditional and just provide a `derive_key_pair` function on the NSS side as well.  You can stub that and have it return `Err(Error::Unsupported)` or something similar to that.

File: ohttp/src/lib.rs [hunk 2]
```diff
@@ -485,4 +506,20 @@ mod test {
         let response2 = client_response2.decapsulate(&enc_response2).unwrap();
         assert_eq!(&response2[..], RESPONSE);
     }
+
+    #[cfg(feature = "rust-hpke")]
+    #[test]
+    fn derive_key_pair() {
```
Comment (martinthomson) [inline ohttp/src/lib.rs:512]: We can make this a known-answer test, right?

File: ohttp/src/lib.rs [hunk 3]
```diff
@@ -116,6 +116,23 @@ impl KeyConfig {
         })
     }
 
+    /// Derive a configuration for the server side from input keying material
```
Comment (martinthomson) [inline ohttp/src/lib.rs:119]: My first thought was that this was supposed to follow: https://www.ietf.org/archive/id/draft-irtf-cfrg-hpke-12.html#name-derivekeypair  Is that right?  Want to cite that?

**Review by martinthomson — COMMENTED** [review]
Seems reasonable.

File: ohttp/src/lib.rs (hunk 1 above)
Comment (chris-wood) [reply ohttp/src/lib.rs:123]: That's better, yeah.

**Comment by chris-wood** [comment]
OK, after some annoying clippy stuff, this should be ready to go.

**Pull request merged**